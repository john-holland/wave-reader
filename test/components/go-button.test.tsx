/**
 * @jest-environment jsdom
 */

import 'jest';
import React from 'react';
import GoButton from '../../src/components/go-button';
import { render, screen } from "@testing-library/react";
import user from "@testing-library/user-event";
import {act} from "react-test-renderer";

test('Link changes the class when hovered', async () => {
    let onGoCalled = false;
    const onGo = () => {
        onGoCalled = true;
    }

    let onStopCalled = false;
    const onStop = () => {
        onStopCalled = true;
    }

    render(
        <GoButton going={false} onStop={onStop} onGo={onGo} />
    );

    const goButtonUnClicked = screen.getByText('go!')
    expect(goButtonUnClicked).toBeTruthy();

    await act(async () => {
        await user.click(goButtonUnClicked);
    });

    expect(onGoCalled).toBeTruthy();

    await act(async () => {
        await user.click(goButtonUnClicked);
    });

    expect(onStopCalled).toBe(true);
    // can't get the testing-library to revalidate the value of the button, but neither could i get the json component
    //  to render with the updated information, so idk.

    // in the meantime...
    // todo: Matrial-UI seems like it suddenly wanted to start fighting with snapshot due to polluting classnames non-deterministically
    expect(goButtonUnClicked).toMatchSnapshot();
});