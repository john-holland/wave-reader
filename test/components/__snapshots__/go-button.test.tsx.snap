// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary Mui-focusVisible MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary sc-bczRLJ iiilXZ css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <h2
      class="sc-gsnTZi llLRgE"
    >
      <span
        class="wave shrink"
      >
        ðŸŒŠ
      </span>
    </h2>
  </span>
  go!
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  >
    <span
      class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
      style="width: 114.49017425089369px; height: 114.49017425089369px; top: -32.245087125446844px; left: -7.245087125446844px;"
    >
      <span
        class="MuiTouchRipple-child"
      />
    </span>
    <span
      class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
      style="width: 86.60254037844386px; height: 86.60254037844386px; top: -18.30127018922193px; left: 6.698729810778069px;"
    >
      <span
        class="MuiTouchRipple-child MuiTouchRipple-childLeaving MuiTouchRipple-childPulsate"
      />
    </span>
  </span>
</button>
`;
